src/Streamly/Internal/System.hs

```diff
@@ -0,0 +1,138 @@
+-- |
+-- Module      : Streamly.Internal.System
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+{-# LANGUAGE TemplateHaskell #-}
+
+module Streamly.Internal.System
+( cmd
+, trim
+)
+where
+
+import Control.Applicative (Alternative(..))
+import Control.Exception (displayException)
+import Data.Functor.Identity (runIdentity)
+import Streamly.Internal.Data.Parser (Parser)
+
+import Language.Haskell.TH
+import Language.Haskell.TH.Quote
+
+import qualified Streamly.Data.Fold as Fold
+import qualified Streamly.Internal.Data.Parser as Parser
+    (some, many, takeWhile1)
+import qualified Streamly.Data.Stream as Stream  (fromList, parse)
+import qualified Streamly.Internal.Unicode.Parser as Parser
+
+data StrSegment
+    = StrText String
+    | StrVar String
+    deriving (Show, Eq)
+
+formatSpace :: String -> String
+formatSpace = foldr go ""
+  where
+    go x acc = x:if x == ' ' then dropWhile (' ' ==) acc else acc
+
+-- | Replace a newline by a space and convert multiple spaces to single space
+--
+-- >>> :set -XQuasiQuotes
+-- >>> import Streamly.Internal.System
+-- >>> trim "  abc   \n   bbb  \n  ccc  "
+-- " abc bbb ccc "
+--
+trim :: String -> String
+trim = formatSpace <$> (unwords . fmap formatSpace . lines)
```

Comment:
You only need `formatSpace` and `trim`.

Reply:
We already have idiomatic code to do this:

```
-- | Replace newlines followed by any number of spaces with a single space.
oneLine :: String -> String
oneLine = unwords . fmap (dropWhile isSpace) . lines
```